### PR TITLE
DPD late match

### PR DESCRIPTION
**Pull request by J-Gras**
Adds the `protocol_late_match` event that allows to track matches of DPD signatures, when the DPD buffer is already exhausted and the match cannot trigger the attachment of an analyzer anymore. A new policy script can be used to add the corresponding information to the `conn.log`.

### CODE REVIEW

**Comment by J-Gras** [comment]
I forgot about the external tests that use the test-all-policy script. Should I update them?

**Comment by JustinAzoff** [comment]
This is a big help for quantifying the issue in https://github.com/zeek/zeek/issues/343

We can increase the dpd buffer size, but there's no easy way to tell if that needs to be done.

File: src/analyzer/protocol/pia/PIA.cc [hunk 1]
```diff
@@ -144,6 +145,20 @@ void PIA_UDP::ActivateAnalyzer(analyzer::Tag tag, const Rule* rule)
...
+		if ( protocol_late_match )
+			{
+			// Queue late match event
+			EnumVal *tval = tag ? tag.AsEnumVal() : GetAnalyzerTag().AsEnumVal();
+			Ref(tval);
+
+			val_list *vl = new val_list;
+			vl->append(BuildConnVal());
+			vl->append(tval);
+			mgr.QueueEvent(protocol_late_match, vl);
```
[the remark this answers is not in the document]
Comment (J-Gras) [reply src/analyzer/protocol/pia/PIA.cc]: Thanks for the hint, I switched to `QueueEventFast`.

**Comment by J-Gras** [comment]
Thanks! Initially I aimed at making that script a package but when I thought about a test case for the event, I ended up with more or less the same script. Given that it may help to tune deployments it might be a good fit to ship it as policy script. However, I am also fine with turning it into a package.

**Comment by jsiwek** [comment]
Per #414, we're generally considering moving policy/ things into external package since they're mostly in the realm of tuning/configuration, so seems preferable to start most new things as external packages rather than move them back out later anyway.  Other opinions?

**Comment by J-Gras** [comment]
In case we keep the policy script I have created [PR](zeek/zeek-testing#3) updating the external test baselines.

**Comment by rsmmr** [comment]
I'm a bit torn where the script should go but given that we haven't settled on a plan/criteria for #414 I wouldn't object merging it in as this point and then reevaluate along with the other scripts.

**Comment by J-Gras** [comment]
If updating the external tests seems overkill but the script should go into policy, excluding it from `test-all-policy.zeek` (comment out) might be the way to go.

**Comment by jsiwek** [comment]
I decided to get this moved along since it's not much more work to re-evaluate along with other `policy/` scripts when we look more into moving things to external packages.

I did comment out the speculative service script from `test-all-policy.zeek` since it changes such a low-level property of the analysis and it seems preferable for the external test suites to run closer to the default configuration (but with things that *add* functionality rather than *change* the low-level functionality further away from what most people would use)

**Pull request merged**